tests/lib/pretty_state.py

```diff
@@ -0,0 +1,72 @@
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from libnmstate.prettystate import PrettyState
+
+TEST_STATE = {
+    "type": "unknown",
+    "name": "foo",
+    "ipv4": {"abc": 1, "enabled": "false"},
+    "slaves": [{"a": 1, "name": "bar1"}, {"a": 2, "name": "bar2"}],
```

Comment:
```suggestion
    "members": [{"a": 1, "name": "bar1"}, {"a": 2, "name": "bar2"}],
```

Since PrettyState is now sorting the state, it would be good to have the input state in a less sorted order. Also test state should be a dictionary with a list of interface states.

Also, there could be some more smaller targetted tests like testing

```yaml
- z
- a
- m
```

or

```yaml
- z: a
  type: test
  name: dictest
- y: b
  name: dictest2
  type: type2 
```

or


```yaml
- x: y
  type: test
  other:
    - 2
    - 1
```

The current test might already cover all the edge cases but using smaller, dedicated tests for edge cases might make it more obvious (not a blocker, just an observation).

Reply:
I agree with @tyll about using "members" instead of "slaves" but I think this is a bigger issue that we need to address in the whole project. I will place an issue for this and take care of it, the issue will be used to discuss which word should we use to replace "master/slave". Thanks!